src/typing/env_js.ml

```diff
@@ -29,6 +29,10 @@ let refinement_key names =
 let is_refinement name =
   (String.length name) >= 5 && (String.sub name 0 5) = "$REFI"
 
+let refined_object_from_key name =
+  let i = String.index name '.' in
+  String.sub name 6 (i - 6)
```

Comment:
In the end, adding a module here seemed like overkill. Instead I was able to combine the `is_refinement` and `refined_object_from_key` functions into a single function, which I also renamed `heap_refinement`.